float64 f_x
float64 f_y
float64 f_z
float64 tau_x
float64 tau_y
float64 tau_z